nav2_costmap_2d/test/integration/test_costmap_2d_publisher.cpp

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2022 Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License. Reserved.
+
+#include <gtest/gtest.h>
+
+#include <future>
+
+#include "nav2_costmap_2d/costmap_2d.hpp"
+#include "nav2_costmap_2d/costmap_2d_publisher.hpp"
+#include "nav2_costmap_2d/costmap_subscriber.hpp"
+#include "nav2_costmap_2d/cost_values.hpp"
+#include "tf2_ros/transform_listener.h"
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+
+class RclCppFixture
+{
+public:
+  RclCppFixture() {rclcpp::init(0, nullptr);}
+  ~RclCppFixture() {rclcpp::shutdown();}
+};
+RclCppFixture g_rclcppfixture;
+
+class TestCostmap2dPublisher : public nav2_util::LifecycleNode
+{
+public:
+  explicit TestCostmap2dPublisher(std::string name)
+  : LifecycleNode(name)
+  {
+    RCLCPP_INFO(get_logger(), "Constructing");
+  }
+
+  ~TestCostmap2dPublisher() {}
+
+  nav2_util::CallbackReturn
+  on_configure(const rclcpp_lifecycle::State &)
+  {
+    RCLCPP_INFO(get_logger(), "Configuring");
+
+    callback_group_ = create_callback_group(
+      rclcpp::CallbackGroupType::MutuallyExclusive, false);
+
+    costmap_ = std::make_shared<nav2_costmap_2d::Costmap2D>(
+      100, 100, 0.05, 0, 0,
+      nav2_costmap_2d::LETHAL_OBSTACLE);
+
+    costmap_pub_ = std::make_shared<nav2_costmap_2d::Costmap2DPublisher>(
+      shared_from_this(),
+      costmap_.get(),
+      "map",
+      "dummy_costmap",
+      true);
+
+    rclcpp::SubscriptionOptions sub_option;
+    sub_option.callback_group = callback_group_;
+
+    std::string topic_name = "/dummy_costmap/static_layer_raw";
```

Comment:
Who's publishing a "raw" topic?

Reply:
And who's giving the static layer a map in the first place for it to be populating with?